src/stream.rs

```diff
@@ -176,9 +180,10 @@ impl RecvBuf {
         Ok(())
     }
 
-    fn pop(&mut self, max_len: usize) -> Result<RangeBuf> {
-        let mut out = RangeBuf::default();
-        let mut out_len = max_len;
+    fn pop(&mut self, out: &mut [u8]) -> Result<(usize, bool)> {
+        let mut fin = false;
+        let mut pop_len = 0;
+        let mut out_len = out.len();
```

Comment:
This variable was misnamed before, so we can probably just rename both `pop_len` and `out_len` as:

`pop_len` => `len`
`out_len` => `cap`

(so it's clear out_len is actually the capacity of the buffer).